### PR TITLE
Better push behavior, listing apps and deleting orgs

### PR DESCRIPTION
**Pull request by viovanov**

### CODE REVIEW

File: cli/paas/client.go [hunk 1]
```diff
@@ -75,13 +79,24 @@ func (c *CarrierClient) Apps() error {
...
-	msg := c.ui.Normal().WithTable("Name", "Status", "Route")
+	msg := c.ui.Normal().WithTable("Name", "Status", "Routes")
 
 	for _, app := range apps {
-		msg = msg.WithTableRow(app.Name)
+		status, err := c.kubeClient.StatefulSetStatus(
+			c.config.EiriniWorkloadsNamespace,
+			fmt.Sprintf("cloudfoundry.org/guid=%s", app.Name))
+		if err != nil {
+			return errors.Wrap(err, "failed to get app status")
```
Comment (andreas-kupries) [inline cli/paas/client.go:89]: I wonder if we should not abort here, but simply show it in the table, in the status cell.
Note also, right now it does not tell the user for which app we got in trouble, and at this point no output was done yet, at all, leaving the user to guess.

File: cli/paas/client.go [hunk 2]
```diff
@@ -75,13 +79,24 @@ func (c *CarrierClient) Apps() error {
...
+		}
+
+		routes, err := c.kubeClient.ListIngressRoutes(
+			c.config.EiriniWorkloadsNamespace,
+			app.Name)
```
Comment (andreas-kupries) [inline cli/paas/client.go:94]: Here we seem to ignore the possible error completely ?
IOW neither showing in the table, nor abort ?

File: cli/paas/client.go [hunk 3]
```diff
@@ -103,6 +118,20 @@ func (c *CarrierClient) CreateOrg(org string) error {
...
+	}
+
+	c.ui.Normal().WithStringValue("name", app).Msg("Deleted app repo.")
+
+	err = c.eiriniClient.EiriniV1().LRPs(c.config.EiriniWorkloadsNamespace).Delete(context.Background(), app, metav1.DeleteOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to delete eirini lrp")
+	}
+
+	c.ui.Normal().WithStringValue("name", app).Msg("Deleted app LRP.")
```
Comment (andreas-kupries) [inline cli/paas/client.go:133]: Should we leak these interna (Repo and LRP) to the user ?
Maybe just tell that the application was deleted ?
In the failures I can see that it is useful to know which component made trouble.
In the success less so.

**Review by andreas-kupries — CHANGES_REQUESTED** [review]
A few questions around error handling in `apps`, and messaging in `delete`.

File: cli/paas/client.go (hunk 1 above)
Comment (viovanov) [reply cli/paas/client.go:89]: This should be an error when we can't talk to the cluster. I think that should be an immediate stop.

Comment (viovanov) [reply cli/paas/client.go:89]: I will include the app name in the error message though

File: cli/paas/client.go (hunk 2 above)
Comment (viovanov) [reply cli/paas/client.go:94]: That's my mistake, thanks!

File: cli/paas/client.go (hunk 3 above)
Comment (viovanov) [reply cli/paas/client.go:133]: Was thinking the same. I'll use more generic terms.

File: cli/paas/client.go (hunk 1 above)
Comment (andreas-kupries) [reply cli/paas/client.go]: Thank you, that should be enough.

**Review by andreas-kupries — APPROVED** [review]

File: cli/cmd/internal/client/apps.go [hunk 4]
```diff
@@ -1 +1,47 @@
...
+
+var ()
+
+// CmdApps implements the carrier app command
+var CmdApps = &cobra.Command{
+	Use:   "apps",
+	Short: "Lists all apps",
+	Args:  cobra.ExactArgs(0),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		app, cleanup, err := paas.BuildApp(cmd.Flags(), nil)
```
Comment (jimmykarily) [inline cli/cmd/internal/client/apps.go:18]: [Minor] The use of "App" in this method's name is a bit confusing (initially I though it was a user's app). We could rename it to `BuildCarrier` or something like that. Could be another PR.

Comment (andreas-kupries) [reply cli/cmd/internal/client/apps.go:18]: Maybe `NewCarrierClient` ? Because that is the main result in the `app` variable, a `CarrierClient` structure.

Comment (jimmykarily) [reply cli/cmd/internal/client/apps.go:18]: :+1:
